framework/Models/PostProcessors/PostProcessorInterface.py

```diff
@@ -0,0 +1,134 @@
+import copy
+import abc
+import os
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from BaseClasses import BaseInterface
+from utils import InputTypes, InputData
+import Runners
+#Internal Modules End--------------------------------------------------------------------------------
+
+class PostProcessorInterface(BaseInterface):
+  """
+    Base class for other postprocessor interfaces (i.e., BasicStatistics, ETImporter).
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    spec = super().getInputSpecification()
+    spec.addParam("subType", InputTypes.StringType, True)
+    return spec
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, runInfoDict, dict, the dictionary containing the runInfo (read in the XML input file)
+      @ Out, None
+    """
+    super().__init__()
+    self.inputCheckInfo  = []     # List of tuple, i.e input objects info [('name','type')]
+    self.action = None            # action
+    self.workingDir = ''          # path for working directory
+    self.printTag = 'PostProcessorInterface'
+    self.outputDataset  = False # True if the user wants to dump the outputs to dataset
+    self.validDataType = ['PointSet','HistorySet'] # The list of accepted types of DataObject
+    ## Currently, we have used both DataObject.addRealization and DataObject.load to
+    ## collect the PostProcessor returned outputs. DataObject.addRealization is used to
+    ## collect single realization, while DataObject.load is used to collect multiple realizations
+    ## However, the DataObject.load can not be directly used to collect single realization
+    ## One possible solution is all postpocessors return a list of realizations, and we only
+    ## use addRealization method to add the collections into the DataObjects
+    self.outputMultipleRealizations = False
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+
+  def initialize(self, runInfo, inputs, initDict=None):
+    """
+      Method to initialize the PostProcessor
+      @ In, runInfo, dict, it is the run info from the jobHandler
+      @ In, inputs, list, it is a list containing whatever is passed with an input role in the step
+      @ In, initDict, dict, optional, dictionary of all objects available in the step is using this model
```

Comment:
no `@Out`

Also, does `initDict` need to be part of the arguments? It doesn't seem to get used. Is this because it's in the argument list of the other postprocessors?

Reply:
added. The "initDict" is used by LimitSurface PP.